docs/chapters/architecture/DLT/diagnostics-log-and-tracing.rst

```diff
@@ -0,0 +1,181 @@
+
+Basic requirements
+""""""""""""""""""
+**Unicode support**
+PELUX default encoding is UTF-8 for all logging messages. GENIVI dlt-daemon works well with UTF-8 as well.
+
+**Dynamic Log adjustments**
+The amount of logs, especially in Trace mode, can be overwhelming. Applications that produce DLT logs should have the functionality of enabling/disabling logs or adjusting the level of log outputs. Ideally each logging call should dynamically check boolean condition and just do nothing if logs are dynamically disabled. It will reduce log traffic inside the system and improve performance in general.
+
+GENIVI DLT provides a back channel from dlt-viewer to applications that generate DLT logs. The user can precisely change log levels for each application individually from the dlt-viewer UI. The user request will reach each application and disable logs in the place they are actually produced.
+
+Quality Goals
+"""""""""""""
+*Reliability*
+Reliability is a key quality for DLT. Developers and testers rely on the logs provided by dlt-daemon to find errors and understand behavior of applications. If the logs will be faulty, there will be a very big struggle to debug the error, or understand the behavior.
+
+Stakeholders
+^^^^^^^^^^^^
+* Software Developers
+* QA Engineers
+
+Constraints
+-----------
+GENIVI's dlt-daemon stores an offline log database on in-RAM file system /tmp/dlt_offlinetrace_20120101_005302.dlt . The size limits for the log files should be adjusted at HU startup based on amount of RAM available. E.g. a Premium HU with 16GB of RAM can use 200MB of offline storage. On the other hand, 4GB of ram uses 50MB of offline storage or less.
+
+Context
+-------
+Business Context
+^^^^^^^^^^^^^^^^
+DLT logs are automotive industry standard. Industry expects any automotive software to support it.
+
+Solution Strategy
+-----------------
+
+Since PELUX is a part of the open source community, we use a component developed by GENIVI.
+
+Below are the open source software that we use from the DLT component.
+
+* `GENIVI dlt-daemon`_
+* `GENIVI dlt-viewer`_
+
+Since GENIVI dlt-daemon supports only C applications, another third party software is required to support C++. Currently, PELUX uses `ivi-logging`_. However, wrapper discussions are still ongoing. We currently have three options which are compared under Design Decisions.
+
+Building blocks view
+--------------------
+
+Level 1
+^^^^^^^
+.. image:: graphs/block-view-lvl1.png
+
+Level 2 
+^^^^^^^
+.. image:: graphs/block-view-lvl2.png
+
+Runtime view
+------------
+Depending on what the developers decide to log and trace, the logs can be viewed via dlt-viewer.
+
+Optionally, Blue PiraT 2 can be configured to work with dlt-daemon, through which multiple gigabytes of logs can be stored.
+
+
+Design decisions
+----------------
+
+C++ wrapper
+^^^^^^^^^^^
+
+`ivi-logging`_
+""""""""""""""
+This component was initially developed by Pelagicore but is not actively maintained at this time.
```

Comment:
Maybe it can be mentioned that this is the solution currently in use, but that we are looking into other solution for [reasons]. And in that case move that text to the top level under "Design decisions" or possibly not include that section at all. In this text we should only mention what solution we currently have and the reasoning behind it.